src/ff/boolean_array.rs

```diff
@@ -0,0 +1,408 @@
+
+            impl ArrayAccess for $name {
+                type Output = Boolean;
+
+                fn get(&self, index: usize) -> Option<Self::Output> {
+                    if index < usize::try_from(<$name>::BITS).unwrap() {
+                        Some(self.0[index].into())
+                    } else {
+                        None
+                    }
+                }
+
+                fn set(&mut self, index: usize, e: Self::Output) {
+                    debug_assert!(index < usize::try_from(<$name>::BITS).unwrap());
+                    self.0.set(index, bool::from(e));
+                }
+            }
+
+            impl SharedValue for $name {
+                type Storage = Store;
+                const BITS: u32 = $bits;
+                const ZERO: Self = Self(<Store>::ZERO);
+            }
+
+            impl Serializable for $name {
+                type Size = <Store as Block>::Size;
+
+                fn serialize(&self, buf: &mut GenericArray<u8, Self::Size>) {
+                    buf.copy_from_slice(self.0.as_raw_slice());
+                }
+
+                fn deserialize(buf: &GenericArray<u8, Self::Size>) -> Self {
+                    Self(<Store>::new(assert_copy(*buf).into()))
+                }
+            }
+
+            impl std::ops::Add for $name {
+                type Output = Self;
+                fn add(self, rhs: Self) -> Self::Output {
+                    Self(self.0 ^ rhs.0)
+                }
+            }
+
+            impl std::ops::AddAssign for $name {
+                fn add_assign(&mut self, rhs: Self) {
+                    *self.0.as_mut_bitslice() ^= rhs.0;
+                }
+            }
+
+            impl std::ops::Sub for $name {
+                type Output = Self;
+                fn sub(self, rhs: Self) -> Self::Output {
+                    self + rhs
+                }
+            }
+
+            impl std::ops::SubAssign for $name {
+                fn sub_assign(&mut self, rhs: Self) {
+                    *self += rhs;
+                }
+            }
+
+            impl std::ops::Neg for $name {
+                type Output = Self;
+                fn neg(self) -> Self::Output {
+                    Self(self.0)
+                }
+            }
+
+            impl Field for $name {
+                const ONE: Self = Self($one);
+
+                fn as_u128(&self) -> u128 {
+                    (*self).into()
+                }
+
+                /// uses hashing in order to be compatible with larger array sizes
+                fn truncate_from<T: Into<u128>>(v: T) -> Self {
+                    let hk = Hkdf::<Sha256>::new(None, &v.into().to_le_bytes());
+                    let mut okm = [0u8; $bytes];
+                    //error invalid length from expand only happens when okm is very large
+                    hk.expand(&[], &mut okm).unwrap();
+                    <$name>::deserialize(&okm.into())
+                }
```

Comment:
The issue is that we initially thought that shared values would be at most 128 bit values so the PRSS creates pseudo uniform 128 bit from which the pseudo uniform shares can be created. For Boolean arrays, we might have much larger values like 256bit, so we need to expand the uniform 128 bit to uniform 256 bits. The hash function here functions like a PRG. Since we already use HKDF, I wanted to reuse it. Alternatively, we could use any cryptographic PRG that expands a 128 bit seed to the size of the shared value.

Reply:
I still do not understand. Let's schedule a 1:1 to talk through this in person.